### PR TITLE
Feat: Edit post and Update to createAsyncThunk

### PR DESCRIPTION
**Pull request by pritam-kr**

### CODE REVIEW

File: src/Components/PostCard/PostCard.jsx [hunk 1]
```diff
@@ -1,12 +1,28 @@
...
+    username,
+    caption,
+    content,
+    likes,
+    _id,
+    avatar,
+    createdAt,
+    updatedAt,
+  } = eachPost || {};
   const likeCount = likes.likeCount;
```
Comment (BelideNikhil) [inline src/Components/PostCard/PostCard.jsx:20]: you can write const { likeCount}  = likes

File: src/Components/PostCard/PostCard.jsx [hunk 2]
```diff
@@ -22,6 +38,14 @@ const PostCard = ({ eachPost, deletedPost, likePost }) => {
     likePost(_id);
   };
 
+  //Post Edit Handler
+
+  const postEditHandler = ({ caption, content }) => {
+  setEditPostModal(true)
+  setPostEditData({content: content, caption: caption, id: _id })
```
Comment (BelideNikhil) [inline src/Components/PostCard/PostCard.jsx:45]: ({content: content, caption: caption, id: _id }) can be written as ({ content, caption, id: _id  }) . It's a shortcut if the key-value pair are same

File: src/Components/PostCard/PostCard.jsx [hunk 3]
```diff
@@ -43,16 +67,16 @@ const PostCard = ({ eachPost, deletedPost, likePost }) => {
           <span>
             <FaIcons.FaEdit
               className="icons profile-icons"
-              onClick={() => postDeleteHandler(_id)}
+              onClick={() =>
+                postEditHandler({ caption: caption, content: content })
```
Comment (BelideNikhil) [inline src/Components/PostCard/PostCard.jsx:71]: same here as well!

File: src/Store/Slice/PostSlice.js [hunk 4]
```diff
@@ -89,19 +186,17 @@ export const {
...
+//     try {
+//       const { data, status } = await axios.get(`/api/posts/user/${username}`);
+//       const posts = data.posts;
+
+//       if (status === 200) {
+//         dispatch(userPosts(posts));
+//       }
+//     } catch (error) {}
+//   };
+// };
```
Comment (BelideNikhil) [inline src/Store/Slice/PostSlice.js:202]: remove this if not needed

**Review by BelideNikhil — APPROVED** [review]
The console logs can be removed
The Code looks clean, nice extraction of services
Great UI as well

File: src/Hooks/usePost.js [hunk 5]
```diff
@@ -1,61 +1,38 @@
...
 export const usePost = () => {
-  const state = useSelector((state) => state);
 
-  const { auth } = state;
-  const token = auth.token;
-  const { userInfo } = auth;
+  const [postData, setPostData] = useState({ content: "", caption: "" });
+ 
+
+  // console.log(editPostData)
```
Comment (Chakravarthi0) [inline src/Hooks/usePost.js:18]: You can remove this console.log(), if not needed

File: src/Pages/Profile/Profile.jsx [hunk 6]
```diff
@@ -1,37 +1,37 @@
...
   const state = useSelector((state) => state);
-
-  const { auth, post } = state;
-  const token = auth.token;
-
-  const { setFollowerModal, setFollowingModal } = useModalContext();
-
+  const { auth:  {userInfo : {username , avatar, firstName, lastName, _id}, token }, post } = state
+  
+  // console.log(state)
```
Comment (Chakravarthi0) [inline src/Pages/Profile/Profile.jsx:22]: You can remove this console.log() as well

**Review by Chakravarthi0 — APPROVED** [review]
clean code.
folder structuring and naming is good.
good to merge

File: src/Components/PostCard/PostCard.jsx (hunk 2 above)
Comment (pritam-kr) [reply src/Components/PostCard/PostCard.jsx:45]: Thank Nikhil

**Pull request merged**